modules/network/wireguard/wireguard.go

```diff
@@ -141,3 +163,28 @@ func newPeer(pubkey, endpoint string, allowedIPs []string) (wgtypes.PeerConfig,
 
 	return peer, nil
 }
+
+func GenerateKey(dir string) (wgtypes.Key, error) {
```

Comment:
hmmm... you have multiple error types emanating from that function, how to handle them ?

Reply:
Well it up to the caller to decide what to do when something fails.